app/helpers/application_helper.rb

```diff
@@ -291,4 +291,34 @@ def role_display_name(role)
     # has been deleted, removed this ternary.
     role.name == "banned" ? "Suspended" : role.name.titlecase
   end
+
+  # Adapted from the following methods:
+  # - cache: https://github.com/rails/rails/blob/d18ada70b398c39cb614c826be91559c2597011c/actionview/lib/action_view/helpers/cache_helper.rb#L166-L175
+  # - write_fragment_for: https://github.com/rails/rails/blob/d18ada70b398c39cb614c826be91559c2597011c/actionview/lib/action_view/helpers/cache_helper.rb#L252-L261
+  def cache_multi(names, cache: Rails.cache, **options, &block)
+    if controller.respond_to?(:perform_caching) && controller.perform_caching
+      name_options = options.slice(:skip_digest)
+      names = names.map { |name| cache_fragment_name(name, **name_options) }
+      content = cache.fetch_multi(*names, options) do |_prefix, key|
+        pos = output_buffer.length
+        yield key
+        output_safe = output_buffer.html_safe?
+        fragment = output_buffer.slice!(pos..-1)
```

Comment:
**suggestion (non-blocking):** We can use an endless range here:

```suggestion
        fragment = output_buffer.slice!(pos..)
```

This has been added in Ruby 2.6, so all versions that don't support this syntax are already EOL.